### PR TITLE
Don't crash if an item has no _fixtureinfo attribute

### LINKED ISSUE #2788: pytest --fixtures-per-test -> 'DoctestItem' object has no attribute '_fixtureinfo'
Hi, I just stumbled over this behaviour when calling 'pytest --fixtures-per-test' in the pytest repo with a fresh venv on 3.6:

pytest --fixtures-per-test

    [...]
    -------------------------------------------------------- fixtures used by test_source_of_class_at_eof_without_newline --------------------------------------------------------
    --------------------------------------------------------------------- (testing/code/test_source.py:395) ----------------------------------------------------------------------
    tmpdir
        Return a temporary directory path object
        which is unique to each test function invocation,
        created as a sub directory of the base temporary
        directory.  The returned object is a `py.path.local`_
        path object.
    tmpdir_factory
        Return a TempdirFactory instance for the test session.
    INTERNALERROR> Traceback (most recent call last):
    INTERNALERROR>   File "/home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/_pytest/main.py", line 110, in wrap_session
    INTERNALERROR>     session.exitstatus = doit(config, session) or 0
    INTERNALERROR>   File "/home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/_pytest/python.py", line 1025, in _show_fixtures_per_test
    INTERNALERROR>     write_item(item)
    INTERNALERROR>   File "/home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/_pytest/python.py", line 1006, in write_item
    INTERNALERROR>     name2fixturedefs = item._fixtureinfo.name2fixturedefs
    INTERNALERROR> AttributeError: 'DoctestItem' object has no attribute '_fixtureinfo'

python --version
    
    Python 3.6.2

    $ uname -av

    Linux ob1 4.12.13-1-ARCH #1 SMP PREEMPT Fri Sep 15 06:36:43 UTC 2017 x86_64 GNU/Linux
    
piplist
    
    Package    Version
    ---------- -------
    pip        9.0.1  
    py         1.4.34 
    pytest     3.2.2  
    setuptools 36.5.0 
    wheel      0.30.0 

pytest --version

    This is pytest version 3.2.2, imported from /home/oliver/.virtualenvs/tox/lib/python3.6/site-packages/pytest.py
    setuptools registered plugins:
      pytest-xdist-1.20.0 at /home/oliver/.virtualenvs/tox/lib/python3.6/site-packages/xdist/plugin.py
      pytest-xdist-1.20.0 at /home/oliver/.virtualenvs/tox/lib/python3.6/site-packages/xdist/looponfail.py
      pytest-timeout-1.2.0 at /home/oliver/.virtualenvs/tox/lib/python3.6/site-packages/pytest_timeout.py
      pytest-forked-0.2 at /home/oliver/.virtualenvs/tox/lib/python3.6/site-packages/pytest_forked/__init__.py
      pytest-cov-2.5.1 at /home/oliver/.virtualenvs/tox/lib/python3.6/site-packages/pytest_cov/plugin.py

### PR DESCRIPTION
**Pull request by obestwalter**
fixes #2788

Simplest possible way of making that error go away. I have obviously no idea what I am doing though.

### CODE REVIEW

File: _pytest/python.py [hunk 1]
```diff
@@ -1003,11 +1003,15 @@ def write_fixture(fixture_def):
             tw.line('    no docstring available', red=True)
 
     def write_item(item):
-        name2fixturedefs = item._fixtureinfo.name2fixturedefs
```
Comment (nicoddemus) [inline _pytest/python.py]: This is just me and a question of preference, but I try to narrow down my try/except blocks as much as possible, so I would write instead:

```python
try:
    name2fixturedefs = item._fixtureinfo.name2fixturedefs
except AttributeError:
    return

if not name2fixturedefs:
    return
```

But really it is up to you. 👍

**Comment by obestwalter** [comment]
Thanks, will fix it up properly tomorrow :)

File: _pytest/python.py (hunk 1 above)
Comment (obestwalter) [reply _pytest/python.py]: good point. will do that.

**Comment by obestwalter** [comment]
Thanks. 

Bit of a mess up now though: I want to learn to have a cleaner commit history so I squashed the changes, but this has close this PR. Need to learn how to do this properly ...

**Comment by obestwalter** [comment]
ah just reopening contains the singel commit then ... *phew*

**Comment by nicoddemus** [comment]
> I want to learn to have a cleaner commit history so I squashed the changes, 

Did you do it locally? I do it frequently and it doesn't need to close the PR. Here's how I usually do it:

```
git rebase -i ... # squash squash squash
git push --force
```

**Comment by obestwalter** [comment]
I did a soft reset to the last commit before my changes, stashed the changes, then did a hard reset to that commit, then a force push and then committed and pushed the changes again ... that was the first thing I could think of how to do this ... Until now I have only really worked on projects, where nobody cares much about git history cleanliness, so I never bothered and stacked up commit on commit - working on tox now though I start to care so I have to up my game there :)

Will do some reading about how to do this the without upsetting the github side of things.

**Comment by obestwalter** [comment]
o.k. the way you did it looks much easier anyway - so:

    git rebase -i origin/master~<wherever I want to go back to> master
    git push origin +master

**Comment by nicoddemus** [comment]
> ...reset to that commit, then a force push...

OK, this is what closed the PR then: this ended up pushing a branch with no changes compared to master. A little weird that this closes the existing PR, but at least it is consistent given that you can't create a PR without any changes compared to the target branch.

**Comment by obestwalter** [comment]
Makes sense.

**Comment by nicoddemus** [comment]
Unless somebody wants more time to review this I will merge this later today! 👍

**Comment by nicoddemus** [comment]
Thanks @obestwalter!

**Pull request merged**